daemon/src/seed.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::{anyhow, bail, Result};
+use bdk::bitcoin::util::bip32::ExtendedPrivKey;
+use bdk::bitcoin::Network;
+use hkdf::Hkdf;
+use rand::Rng;
+use sha2::Sha256;
+use std::convert::TryInto;
+use std::path::Path;
+
+pub struct Seed([u8; 256]);
+
+impl Seed {
+    /// Initialize a [`Seed`] from a path.
+    ///
+    /// Fails if the file does not exist or it exists but would be overwritten.
+    pub async fn initialize(seed_file: &Path, generate: bool) -> Result<Seed> {
+        let exists = seed_file.exists();
+
+        let seed = match (exists, generate) {
```

Comment:
out of curiosity: would placing `seed_file.exists()` here directly cause any difference, e.g. would that we evaluated multiple times when trying to match different arms?

Reply:
Nope, would only be evaluated once AFAIK!

It just liked the symmetry of matching against two variables instead of one variable and one expression.